klippy/extras/fan.py

```diff
@@ -28,9 +30,22 @@ def __init__(self, config, default_shutdown_speed=0.):
         self.mcu_fan.setup_cycle_time(cycle_time, hardware_pwm)
         shutdown_power = max(0., min(self.max_power, shutdown_speed))
         self.mcu_fan.setup_start_value(0., shutdown_power)
+
+        # Setup tachometer
+        tach_pin = config.get('tachometer_pin', None)
+        if tach_pin is not None:
+            self.tach_ppr = config.getint('tachometer_ppr', 2, minval=1)
+            POLL_TIME = 1. / 4000 # enough for 60000 rpm max at 2 ppr
+            SAMPLE_TIME = 1.
+            tach_counter = pulse_counter.FrequencyCounter(
+                self.printer, tach_pin, SAMPLE_TIME, POLL_TIME)
```

Comment:
FWIW, I think it would be worthwhile to add a `config.getfloat('tachometer_poll_interval`, .002)` config option and default it to a conservative (not very high) value.  I'd guess most fans aren't super fast and I would guess most users would prefer a conservative value that definitely wont impact their stepping rates.